### ISSUE TITLE
Getting data pointer of Matrix4f?

### ISSUE THREAD

**Issue by lazysquid** [stereolabs/zed-examples#58]
Hello. Currently I'm trying to copy tracking pose information(transformation matrix) of each frame to *Eigen::Matrix4f*.
To do that, I should know data pointer of *sl::Matrix4f*.
In ZED online api document, it doesn't mention anything about it.
However I found this in *types.hpp* of ZED SDK

```cpp
    \class Matrix4f
    \ingroup Core_group
    \brief Represents a generic fourth-dimensional matrix.

    sIt is defined in a row-major order, it means that, in the value buffer, the entire first row is stored first, followed by the entire second row, and so on.
    You can access the data by the 'm' ptr or by the element attribute.
    \f[\begin{bmatrix}
    r00 & r01 & r02 & tx \\
    r10 & r11 & r22 & ty \\
    r20 & r21 & r22 & tz \\
    m30 & m31 & m32 & m33
    \end{bmatrix} \f]

.... 

// in Matrix4f class 
        union {

            struct {
                float r00, r01, r02, tx, r10, r11, r12, ty, r20, r21, r22, tz, m30, m31, m32, m33;
            };
            // ptr to inner data.
            float m[nbElem];
        };
        /// @endcond

.... 

```
So what I want to ask is that even though public array pointer *m* does not appeared in ZED api doc, am I okay to use it?

Also, is there any handy way to get globally registered point cloud?
Right now, I just copy point cloud of single frame and multiply inverse of pose transformation in every single point.
But this is quite cumbersome and this feature seems to be implemented quite efficiently in ZED SDK.
Do you guys have any plans for implementing it in future ZED SDK?

**Comment by WASCHMASCHINE** [comment]
And how is this transformation cumbersome? It is a clean way to separate the point cloud from the ESTIMATED transformation, and it would be literally only ONE additional matrix multiplication per frame, since you use a transformation chain anyway for display. Again, that is not per point, per FRAME.

**Comment by lazysquid** [comment]
https://www.stereolabs.com/developers/documentation/API/v2.4.0/classsl_1_1Transform.html 
As I said above the m pointer is not mentioned in latest api document. Please check it out.

**Comment by lazysquid** [comment]
Mathematically you are right.
As far as I know zed only provide point cloud defined in each frame right? They are already defined in that frame so that there is no way to avoid  mulitplication of transformation. 
Does zed provide a way to apply the transformation to frame? Am I missing something?

**Comment by adujardin** [comment]
Hi,
Not sure if I correctly understood what you wanted but you can already get the point clouds in world reference frame by changing the [RuntimeParameters](https://www.stereolabs.com/developers/documentation/API/v2.4.0/structsl_1_1RuntimeParameters.html) given to the `grab` function.
`measure3D_reference_frame` should be put to [REFERENCE_FRAME_WORLD](https://www.stereolabs.com/developers/documentation/API/v2.4.0/group__PositionalTracking__group.html#ga7160d24146fdddd5a7e8f67fde1c4487).

If you want to cast the sl::Matrix4f to Eigen::Matrix4f, feel free to use the pointer `m`, like [this](https://gist.github.com/adujardin/8133a0c49317b18fb3545f76c1bec5db).

**Comment by WASCHMASCHINE** [comment]
-> 2.4.0 Matrix4f
https://www.stereolabs.com/developers/documentation/API/v2.4.0/classsl_1_1Matrix4f.html

Detailed Description

Represents a generic fourth-dimensional matrix.

sIt is defined in a row-major order, it means that, in the value buffer, the entire first row is stored first, followed by the entire second row, and so on. You can access the data by the 'm' ptr or by the element attribute.

\[\begin{bmatrix} r00 & r01 & r02 & tx \\ r10 & r11 & r22 & ty \\ r20 & r21 & r22 & tz \\ m30 & m31 & m32 & m33 \end{bmatrix} \]

**Comment by lazysquid** [comment]
@WASCHMASCHINE 
So it was in Matrix4f... I only checked Transform page. Sorry about that. It was my dumb mistake.

@adujardin  @WASCHMASCHINE 
I had no problem using the pointer `m`. I worked with this code snippet and it works nicely.
```cpp
sl::Matrix4f zed_transform = pose.pose_data;
T = Eigen::Map<Eigen::Matrix<float, 4, 4, Eigen::RowMajor>>(zed_transform.m);
```
Thanks for sharing code snippet anyway.

What I wanted to say in this issue was in latest api document of Transform `m` is not written and wanted to let you guys know. But it details were in Matrix4f. 

Also

> Not sure if I correctly understood what you wanted but you can already get the point clouds in world reference frame by changing the RuntimeParameters given to the grab function.
measure3D_reference_frame should be put to REFERENCE_FRAME_WORLD.
Yes this is exactly what I wanted.


@adujardin So you are saying that 
If I use `REFERENCE_FRAME_WORLD`, I can get points registered in world frame?


I already use that parameter however I thought it only affect to tracking pose and point cloud just stay in camera frame. This is the reason why I thought like that.

> REFERENCE_FRAME_WORLD  | The transform of sl::Pose will contains the motion with reference to the world frame (previously called PATH).


So I tried to transform all point with inverse transform of corresponding pose.

But clearly there is another api doc which describe

>Provides 3D measures (point cloud and normals) in the desired reference frame (default is REFERENCE_FRAME_CAMERA) 
default : REFERENCE_FRAME_CAMERA

If I can get globally registered point cloud by only change the parameter that will be best for me. Now I can understand what @WASCHMASCHINE said. 

Thank you guys. And sorry about dumb mistakes.
